### PR TITLE
[Helix] run on arm queues

### PR DESCRIPTION
**Pull request by HaoK**
Cleaned up for rebase version of https://github.com/aspnet/AspNetCore/pull/8844

### CODE REVIEW

File: eng/targets/Helix.Common.props [hunk 1]
```diff
@@ -11,20 +12,30 @@
...
-    <HelixAvailableTargetQueue Include="Debian.9.Amd64.Open" Platform="Linux" />
-    <HelixAvailableTargetQueue Include="Redhat.7.Amd64.Open" Platform="Linux" />
-    <HelixAvailableTargetQueue Include="(Fedora.28.Amd64.Open)Ubuntu.1604.Amd64.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:fedora-28-helix-09ca40b-20190508143249" Platform="Linux" />
-
-    <!--  TODO: re-enable Debian.9.Arm64.Open and Ubuntu.1804.Arm64.Open    -->
+    <HelixAvailableArch Include="x64" />
+    <HelixAvailableArch Include="arm64" />
+    
+    <!-- x64 queues -->
+    <HelixAvailableTargetQueue Include="Windows.10.Amd64.ClientRS4.VS2017.Open" Platform="Windows" Condition="'$(TargetArchitecture)' == 'x64'"/>
```
Comment (HaoK) [inline eng/targets/Helix.Common.props]: @aspnet/build do you guys have any suggestions for better msbuild magic to accomplish this filtering for arm vs x64?

File: src/Security/test/AuthSamples.FunctionalTests/AuthSamples.FunctionalTests.csproj [hunk 2]
```diff
@@ -46,34 +51,44 @@
     </ItemGroup>
     <Copy
       SourceFiles="@(_PublishFiles)"
-      DestinationFolder="$(PublishDir)" />
+      DestinationFolder="$(PublishDir)"
+      ContinueOnError="true" />
```
Comment (dougbu) [inline src/Security/test/AuthSamples.FunctionalTests/AuthSamples.FunctionalTests.csproj]: Same question about failing fast.

File: src/Identity/test/Identity.FunctionalTests/Microsoft.AspNetCore.Identity.FunctionalTests.csproj [hunk 3]
```diff
@@ -34,19 +37,23 @@
     </ItemGroup>
     <Copy
       SourceFiles="@(_PublishFiles)"
-      DestinationFolder="$(PublishDir)" />
+      DestinationFolder="$(PublishDir)"
+      ContinueOnError="true" />
```
Comment (dougbu) [inline src/Identity/test/Identity.FunctionalTests/Microsoft.AspNetCore.Identity.FunctionalTests.csproj]: Same question about failing fast.

File: src/Azure/AzureAD/test/FunctionalTests/Microsoft.AspNetCore.Authentication.AzureAD.FunctionalTests.csproj [hunk 4]
```diff
@@ -23,18 +26,21 @@
     </ItemGroup>
     <Copy
       SourceFiles="@(_PublishFiles)"
-      DestinationFolder="$(PublishDir)" />
+      DestinationFolder="$(PublishDir)"
+      ContinueOnError="true" />
```
Comment (dougbu) [inline src/Azure/AzureAD/test/FunctionalTests/Microsoft.AspNetCore.Authentication.AzureAD.FunctionalTests.csproj]: This looks odd. Won't the tests just waste time and fail later if assets are incomplete? Seems better to fail fast.

File: eng/targets/Helix.Common.props (hunk 1 above)
Comment (dougbu) [reply eng/targets/Helix.Common.props]: Seems reasonable as-is. What would you like to improve?

Comment (HaoK) [reply eng/targets/Helix.Common.props]: Seems different than the other attribute pattern used for platform and enablebydefault, I guess maybe I can just move each arch into its own item group with a condition which will make it feel less clunky.  Just wasn't sure if there was a better pattern to do something like this

File: src/Security/test/AuthSamples.FunctionalTests/AuthSamples.FunctionalTests.csproj (hunk 2 above)
Comment (HaoK) [reply src/Security/test/AuthSamples.FunctionalTests/AuthSamples.FunctionalTests.csproj]: I can back these out for now, I'm not sure why these projects are failing with copy file exists errors on the arm64 builds, these were trying to work around those errors, they are skipped for now on arm64

File: src/Azure/AzureAD/test/FunctionalTests/Microsoft.AspNetCore.Authentication.AzureAD.FunctionalTests.csproj (hunk 4 above)
Comment (HaoK) [reply src/Azure/AzureAD/test/FunctionalTests/Microsoft.AspNetCore.Authentication.AzureAD.FunctionalTests.csproj]: This error is actually typically a double copy failure

**Comment by HaoK** [comment]
Finally looks like helix tests are passing on arm64, marking as ready for review

File: eng/targets/Helix.Common.props [hunk 5]
```diff
@@ -23,8 +27,16 @@
...
     <HelixAvailableTargetQueue Include="(Fedora.28.Amd64.Open)Ubuntu.1604.Amd64.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:fedora-28-helix-09ca40b-20190508143249" Platform="Linux" />
-
-    <!--  TODO: re-enable Debian.9.Arm64.Open and Ubuntu.1804.Arm64.Open    -->
+  </ItemGroup>
+    
+  <ItemGroup Condition="'$(IsWindowsOnlyTest)' != 'true' AND '$(TargetArchitecture)' == 'arm64'">
+    <!-- arm64 queues -->
+    <HelixAvailableTargetQueue Include="(Debian.9.Arm64.Open)Ubuntu.1604.Arm64.Docker.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:debian-9-helix-arm64v8-0a0ebdd-20190312215438" Platform="Linux" />
+    
+    <!-- Need to resolve permission issues on this docker queue
```
Comment (dougbu) [inline eng/targets/Helix.Common.props]: Do we have an issue tracking this TODO?

File: eng/targets/Helix.Common.props [hunk 6]
```diff
@@ -10,7 +10,11 @@
     <HelixAvailablePlatform Include="Windows" />
     <HelixAvailablePlatform Include="OSX" />
     <HelixAvailablePlatform Include="Linux" />
-
+  </ItemGroup>
+    
+    <!-- x64 queues -->
+  <ItemGroup Condition="'$(IsWindowsOnlyTest)' != 'true' AND '$(TargetArchitecture)' == 'x64'">
+
```
Comment (dougbu) [inline eng/targets/Helix.Common.props]: nit: No need for blank lines after an open tag

File: src/Identity/test/Identity.FunctionalTests/Microsoft.AspNetCore.Identity.FunctionalTests.csproj [hunk 7]
```diff
@@ -37,7 +40,7 @@
       DestinationFolder="$(PublishDir)" />
     <Copy
       SourceFiles="@(_PagesFiles)"
-      DestinationFolder="$(PublishDir)\Identity.DefaultUI.WebSite\Pages" />
+      DestinationFolder="$(PublishDir)\Identity.DefaultUI.WebSite\Pages"/>
```
Comment (dougbu) [inline src/Identity/test/Identity.FunctionalTests/Microsoft.AspNetCore.Identity.FunctionalTests.csproj]: nit: Our usual formatting includes the space before `/>`

File: src/Security/test/AuthSamples.FunctionalTests/AuthSamples.FunctionalTests.csproj [hunk 8]
```diff
@@ -46,13 +48,13 @@
     </ItemGroup>
     <Copy
       SourceFiles="@(_PublishFiles)"
-      DestinationFolder="$(PublishDir)" />
+      DestinationFolder="$(PublishDir)"  />
```
Comment (dougbu) [inline src/Security/test/AuthSamples.FunctionalTests/AuthSamples.FunctionalTests.csproj]: nit: Our usual formatting uses just one space before `/>`

**Review by dougbu — APPROVED** [review]
Changes look good but Helix run did not pass. Once it does, feel free to merge

**Comment by HaoK** [comment]
@dougbu I see one test failure that looks like a pass on rerun, https://mc.dot.net/#/user/aspnetcore/pr~2Faspnet~2Faspnetcore/ci/20190626.58.arm64.1 shows all green

https://mc.dot.net/#/user/aspnetcore/pr~2Faspnet~2Faspnetcore/ci/20190626.58.arm64.1/workItem/dotnet-user-secrets.Tests~2Fnetcoreapp3.0/wilogs took a second try

**Comment by dougbu** [comment]
Microsoft.Extensions.SecretManager.Tools.Tests.Microsoft.Extensions.SecretManager.Tools.Tests.SecretManagerTests.SetSecret_Update_Existing_Secret shows as failing in https://dev.azure.com/dnceng/public/_build/results?buildId=240998&view=ms.vss-test-web.build-test-results-tab despite what https://mc.dot.net/#/user/aspnetcore/pr~2Faspnet~2Faspnetcore/ci/20190626.58.arm64.1 indicates

File: eng/targets/Helix.Common.props (hunk 5 above)
Comment (HaoK) [reply eng/targets/Helix.Common.props]: https://github.com/aspnet/AspNetCore/issues/11718

**Comment by HaoK** [comment]
All green this time, merging

**Pull request merged**